app/src/main/java/com/kickstarter/ui/fragments/RewardsFragment.kt

```diff
@@ -0,0 +1,65 @@
+package com.kickstarter.ui.fragments
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.PagerSnapHelper
+import com.kickstarter.R
+import com.kickstarter.libs.BaseFragment
+import com.kickstarter.libs.qualifiers.RequiresFragmentViewModel
+import com.kickstarter.libs.rx.transformers.Transformers
+import com.kickstarter.libs.utils.RewardDecoration
+import com.kickstarter.models.Project
+import com.kickstarter.ui.adapters.HorizontalRewardsAdapter
+import com.kickstarter.viewmodels.RewardFragmentViewModel
+import kotlinx.android.synthetic.main.fragment_rewards.*
+
+@RequiresFragmentViewModel(RewardFragmentViewModel.ViewModel::class)
+class RewardsFragment : BaseFragment<RewardFragmentViewModel.ViewModel>() {
+
+    private var rewardsAdapter = HorizontalRewardsAdapter()
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        super.onCreateView(inflater, container, savedInstanceState)
+        return inflater.inflate(R.layout.fragment_rewards, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupRecyclerView()
+
+        this.viewModel.outputs.project()
+                .compose(bindToLifecycle())
+                .compose(Transformers.observeForUI())
+                .subscribe { rewardsAdapter.populateRewards(it) }
+
+    }
+
+    private fun setupRecyclerView() {
+        rewards_recycler.adapter = rewardsAdapter
+        rewards_recycler.layoutManager = LinearLayoutManager(activity, LinearLayoutManager.HORIZONTAL, false)
+        addItemDecorator()
+        addSnapHelper()
+    }
+
+    private fun addSnapHelper() {
+        val snapHelper = PagerSnapHelper()
+        snapHelper.attachToRecyclerView(rewards_recycler)
+    }
+
+    private fun addItemDecorator() {
+        val radius = resources.getDimensionPixelSize(R.dimen.circle_radius).toFloat()
+        val inactiveColor = ContextCompat.getColor(rewards_recycler.context, R.color.ksr_dark_grey_400)
+        val activeColor = ContextCompat.getColor(rewards_recycler.context, R.color.ksr_soft_black)
+        val margin = resources.getDimension(R.dimen.reward_margin).toInt()
+        val padding = radius * 2
+        rewards_recycler.addItemDecoration(RewardDecoration(margin, activeColor, inactiveColor, radius, padding))
+    }
+
+    fun takeProject(project: Project) {
```

Comment:
Let's move the public methods above the privates ones and keep them alphabetical.